sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultPlanGenerator.java

```diff
@@ -59,6 +59,11 @@ protected Phase from(RawPhase rawPhase, String phaseName, Collection<PodSpec> po
                 List<String> taskNames = taskSpecs.stream()
                         .map(taskSpec -> taskSpec.getName())
                         .collect(Collectors.toList());
+
+                if (!CollectionUtils.isEmpty(rawPhase.getTasks())) {
```

Comment:
Is this supposed to be a `taskNames.addAll()`? Why/why not? (could include a comment in either case)